packages/javascript/bh-shared-ui/src/views/PrivilegeZones/History/FilterDialog/FilterDialog.tsx

```diff
@@ -239,96 +264,115 @@ const FilterDialog: FC<{
-                                                ) : (
-                                                    <SelectContent>
-                                                        <SelectItem value={'SYSTEM'}>SYSTEM</SelectItem>
-                                                        {bloodHoundUsersQuery.data
-                                                            ?.filter((user) => {
-                                                                let hasPermission = false;
-                                                                user.roles.forEach((role) => {
-                                                                    role.permissions.forEach((permission) => {
-                                                                        if (
-                                                                            permission.name === 'ManageUsers' &&
-                                                                            permission.authority === 'auth'
-                                                                        )
-                                                                            hasPermission = true;
-                                                                    });
-                                                                });
-
-                                                                return hasPermission;
-                                                            })
-                                                            .sort((a, b) =>
-                                                                (a.email_address || a.principal_name).localeCompare(
-                                                                    b.email_address || b.principal_name
+                                            <Button
+                                                variant={'text'}
+                                                disabled={!field.value}
+                                                className={cn('w-1/12 p-0', { invisible: !field.value })}
+                                                onClick={() => {
+                                                    form.setValue(field.name, '');
+                                                }}>
+                                                <FontAwesomeIcon icon={faClose} />
+                                            </Button>
+                                        </div>
+                                        {bloodHoundUsersQuery.isLoading ? (
+                                            <Skeleton className='h-10 w-24' />
+                                        ) : (
+                                            <SelectContent>
+                                                <SelectItem value={SystemString}>{SystemString}</SelectItem>
+                                                {bloodHoundUsersQuery.data
+                                                    ?.filter((user) => {
```

Comment:
suggestion: I think a useMemo is merited here due to the level of filtering / looping involved